classes/migration/upgrade/v3_4_0/UpgradeMigration.inc.php

```diff
@@ -80,6 +86,67 @@ public function up(): void
                 $table->dropColumn('setting_type');
             });
         }
+
+        // pkp/pkp-lib#6073: Delete SubmissionFiles/Notes entries that correspond to nonexistent queries.
+        $orphanedIds = DB::table('notes AS n')
+            ->leftJoin('queries AS q', 'n.assoc_id', '=', 'q.query_id')
+            ->where('n.assoc_type', '=', self::ASSOC_TYPE_QUERY)
+            ->whereNull('q.query_id')
+            ->pluck('n.note_id', 'n.assoc_id');
+
+        foreach ($orphanedIds as $neQueryId => $noteId) {
+            error_log("Removing submission files that relates to the entry ID ${noteId} which will be deleted as orphan object");
+            $notesFileRows = DB::table('submission_files as sf')
+                ->join('files as f', 'sf.file_id', '=', 'f.file_id')
+                ->where('sf.assoc_id', '=', $noteId)
+                ->where('sf.file_stage', '=', self::SUBMISSION_FILE_QUERY)
+                ->get([
+                        'sf.submission_file_id as submissionFileId',
+                        'sf.file_id as fileId',
+                        'f.path as filePath'
+                    ]);
+
+            foreach ($notesFileRows as $submissionFileRow) {
+                $submissionFileId = $submissionFileRow->submissionFileId;
+                $submissionFileFileId = $submissionFileRow->fileId;
+                $submissionFilePath = $submissionFileRow->filePath;
+
+                error_log("Removing submission file revisions that relate to the submission file entry ID ${submissionFileId} which will be deleted.");
+                DB::table('submission_file_revisions')
+                    ->where('submission_file_id', '=', $submissionFileId)
+                    ->delete();
+
+                error_log("Removing review round files that relate to the submission file entry ID ${submissionFileId} which will be deleted.");
+                DB::table('review_round_files')
+                    ->where('submission_file_id', '=', $submissionFileId)
+                    ->delete();
+
+                error_log("Removing review files that relate to the submission file entry ID ${submissionFileId} which will be deleted.");
+                DB::table('review_files')
+                    ->where('submission_file_id', '=', $submissionFileId)
+                    ->delete();
+
+                // Create entry in files and revisions tables for every submission_file
+                $fileService = Services::get('file');
```

Comment:
I would avoid using the file service. This is likely to change implementation, which will probably break this migration in future releases. Best to to go the database and filesystem directly in migration classes.

Reply:
Changed. 

Not sure whether I used the correct code for this, though. I copied the code from the `PKPFileService` constructor.

Reply:
It looks OK -- if it's tested, please mark this conversation resolved!

Reply:
I changed the code a little to better handle possible errors on the files deletion. I executed
`php .\tools\migration.php  PKP\migration\upgrade\v3_4_0\I8073_RemoveNotesWithoutQueriesAndRelatedObjects up` and it seems to be working as it supposed to.